### PR TITLE
Gives the RCD a unique suicide act (well, 2 or 4, depending on how you look at it)

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
The RCD now has a suicide act that changes depending on the RCD's current mode.

Deconstruction will deconstruct you.

Walls, airlocks, and windows will build the aforementioned structure inside you, gibbing you (yes, it will ACTUALLY be built).

If you do not have enough ammo to commit suicide, you shall be overcome with a wave of shame for not even bothering to load compressed matter into it.

## Why It's Good For The Game

Yes.

## Testing
An RCD was given to a group of suicidal skrell, who one-by-one proceeded to use the device to end themselves in various ways. Except for the last skrell. They didn't have enough compressed matter. They were overcome by SHAME.

One tried to walk away mid-suicide. Couldn't stop it.

Another tried dropping the RCD. They couldn't.

## Changelog
:cl:
add: RCD now has a suicide act that changes based on the selected mode.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/RCD.dm [hunk 1]
```diff
@@ -131,6 +131,66 @@
...
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/user)
+	var/turf/suicide_tile = get_turf(src)
+	if(src.mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger.  It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, 1)
+		to_chat(user, "Deconstructing User...")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		sleep(5 SECONDS)
```
Comment (AyIong) [inline code/game/objects/items/weapons/RCD.dm:141]: Wait, that's illegal

Comment (DGamerL) [reply code/game/objects/items/weapons/RCD.dm:141]: Yeah why is this not just doing a `do_after` lmao

File: code/game/objects/items/weapons/RCD.dm [hunk 2]
```diff
@@ -131,6 +131,66 @@
...
+
+	if(src.mode == MODE_AIRLOCK && checkResource(10, user))
+		. = mode_airlock(suicide_tile, user)
+		if(user.l_hand == src || user.r_hand == src)
+			user.visible_message("<span class='suicide'>[src] creates an airlock inside [user], causing [user.p_them()] to explode!</span>")
+			user.gib()
+			return OBLITERATION
+		user.visible_message("<span class='suicide'>[user] doesn't have [src] anymore, [user.p_they()] cannot commit suicide with it!</span>", \
+		"<span class='suicide'>You cannot commit suicide without [src]!</span>")
+		return
```
Comment (lewcc) [inline code/game/objects/items/weapons/RCD.dm:186]: Fair bit of duplicated here

File: code/game/objects/items/weapons/RCD.dm [hunk 3]
```diff
@@ -536,7 +596,7 @@
 /**
  * Called in `afterattack()` if `mode` is set to `MODE_WINDOW`.
  *
- * Constructs a grille and 4 reinforced window panes at the given location `A`.
+ * Constructs a grille a fulltile reinforced window at the given location `A`.
```
Comment (lewcc) [inline code/game/objects/items/weapons/RCD.dm:599]: the good ol' days huh

**Review by DGamerL — COMMENTED** [review]
Three quick things
1. All mentions of `src.mode` can be replaced with `mode`
2. All returns when the suicide act fails should return `SHAME` instead of what results out of the RCD proc
3. No need to do `. = [RCD_proc_here]"`, you should be able to just call it normally as we are not using the return value from that RCD proc at all

**Comment by Fordoxia** [comment]
> Three quick things
> 
> 1. All mentions of `src.mode` can be replaced with `mode`
> 2. All returns when the suicide act fails should return `SHAME` instead of what results out of the RCD proc
> 3. No need to do `. = [RCD_proc_here]"`, you should be able to just call it normally as we are not using the return value from that RCD proc at all

All done.

File: code/game/objects/items/weapons/RCD.dm [hunk 4]
```diff
@@ -131,6 +131,60 @@
...
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger.  It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "Deconstructing User...")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))
+			return
+		QDEL_NULL(suicide_A)
+		if(user.l_hand == src || user.r_hand == src)	// Do not commit die if the RCD isn't in your hands.
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:146]: Can you make this an early return?

File: code/game/objects/items/weapons/RCD.dm [hunk 5]
```diff
@@ -131,6 +131,60 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags = NODROP				// You cannot drop. You commit to die.
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:136]:
```suggestion
	flags |= NODROP				// You cannot drop. You commit to die.
```

File: code/game/objects/items/weapons/RCD.dm [hunk 6]
```diff
@@ -131,6 +131,60 @@
...
+			for(var/obj/item/W in user)	// Do not delete all their stuff.
+				user.unEquip(W)
+			user.dust()
+			return OBLITERATION
+		return SHAME
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger.  It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	if(mode == MODE_TURF && checkResource(3, user))		// Check you can afford to build this.
+		mode_turf(suicide_tile, user)					// Build it.
+		if(user.l_hand == src || user.r_hand == src)	// Do not commit die if the RCD isn't in your hands.
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:159]: Same here

File: code/game/objects/items/weapons/RCD.dm [hunk 7]
```diff
@@ -131,6 +131,60 @@
...
+			user.visible_message("<span class='suicide'>[src] creates a wall inside [user], causing [user.p_them()] to explode!</span>")
+			user.gib()
+			return OBLITERATION
+		return SHAME
+
+	if(mode == MODE_WINDOW && checkResource(2, user))
+		mode_window(suicide_tile, user)
+		if(user.l_hand == src || user.r_hand == src)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:167]: And here

File: code/game/objects/items/weapons/RCD.dm [hunk 8]
```diff
@@ -131,6 +131,60 @@
...
+			user.visible_message("<span class='suicide'>[src] creates a window inside [user], causing [user.p_them()] to explode!</span>")
+			user.gib()
+			return OBLITERATION
+		return SHAME
+
+	if(mode == MODE_AIRLOCK && checkResource(10, user))
+		mode_airlock(suicide_tile, user)
+		if(user.l_hand == src || user.r_hand == src)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:175]: Here too

File: code/game/objects/items/weapons/RCD.dm [hunk 9]
```diff
@@ -131,6 +131,61 @@
...
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger.  It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "Deconstructing User...")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))					// Deconstruction of most structures takes 5 seconds, wait for the animation to finish.
+			return
+		QDEL_NULL(suicide_A)
+		if(!user.l_hand == src && !user.r_hand == src)	// Do not commit die if the RCD isn't in your hands.
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:146]:
```suggestion
		if(!(user.l_hand == src) && !(user.r_hand == src))	// Do not commit die if the RCD isn't in your hands.
```

File: code/game/objects/items/weapons/RCD.dm [hunk 10]
```diff
@@ -131,6 +131,61 @@
...
+		useResource(5, user)	// Consume ammo.
+		user.visible_message("<span class='suicide'>[user] deconstructs [user.p_they()]self with [src]!</span>")
+		playsound(loc, usesound, 50, TRUE)
+		for(var/obj/item/W in user)	// Do not delete all their stuff.
+			user.unEquip(W)			// Dump everything on the floor instead.
+		flags &= ~NODROP			// NODROP must be removed so the RCD doesn't get dusted along with them. Having this come after the unequipping puts the RCD on top of the pile of stuff (held items fall to the floor when dusting).
+		user.dust()
+		return OBLITERATION
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger.  It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:157]:
```suggestion
	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
```

File: code/game/objects/items/weapons/RCD.dm [hunk 11]
```diff
@@ -131,6 +131,61 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger.  It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:139]:
```suggestion
		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
```

File: code/game/objects/items/weapons/RCD.dm [hunk 12]
```diff
@@ -131,6 +131,61 @@
...
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "Deconstructing User...")
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:141]: No span class?

Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm:141]: All of the other construction/deconstruction dialog lacks span classes.

Should I give all of them spans or..?

Comment (Henri215) [reply code/game/objects/items/weapons/RCD.dm:141]: Only here is good.

File: code/game/objects/items/weapons/RCD.dm [hunk 13]
```diff
@@ -131,6 +131,61 @@
...
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "Deconstructing User...")
```
Comment (Fordoxia) [inline code/game/objects/items/weapons/RCD.dm:141]:
```suggestion
		to_chat(user, "<span class='notice'>Deconstructing User...</span>")
```

File: code/game/objects/items/weapons/RCD.dm [hunk 14]
```diff
@@ -131,6 +131,61 @@
...
+		useResource(5, user)	// Consume ammo.
+		user.visible_message("<span class='suicide'>[user] deconstructs [user.p_they()]self with [src]!</span>")
+		playsound(loc, usesound, 50, TRUE)
+		for(var/obj/item/W in user)	// Do not delete all their stuff.
+			user.unEquip(W)			// Dump everything on the floor instead.
+		flags &= ~NODROP			// NODROP must be removed so the RCD doesn't get dusted along with them. Having this come after the unequipping puts the RCD on top of the pile of stuff (held items fall to the floor when dusting).
+		user.dust()
+		return OBLITERATION
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:157]: This shouldnt be before the first suicide method?

File: code/game/objects/items/weapons/RCD.dm [hunk 15]
```diff
@@ -131,6 +131,61 @@
...
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='notice'>Deconstructing User...</span>")
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:141]:
```suggestion
		to_chat(user, "<span class='notice'>Deconstructing user...</span>")
```

File: code/game/objects/items/weapons/RCD.dm (hunk 14 above)
Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm:157]: First suicide method has its own text where they point the RCD at their chest to deconstruct themselves.

All other suicide methods involving construction have the RCD stuck in the mouth so the constructed object appears inside the user (allowing for them to explode from having a structure suddenly appearing inside their body).

File: code/game/objects/items/weapons/RCD.dm [hunk 16]
```diff
@@ -131,6 +131,61 @@
...
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='notice'>Deconstructing user...</span>")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))					// Deconstruction of most structures takes 5 seconds, wait for the animation to finish.
+			return
+		QDEL_NULL(suicide_A)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:145]: Could you make the `do_after` 6,1 seconds?
And you should put the QDEL in the `if`-statement

File: code/game/objects/items/weapons/RCD.dm [hunk 17]
```diff
@@ -131,6 +131,61 @@
...
+
+	if(mode == MODE_AIRLOCK && checkResource(10, user))
+		mode_airlock(suicide_tile, user)
+		if(!user.l_hand == src && !user.r_hand == src)
+			return SHAME
+		user.visible_message("<span class='suicide'>[src] creates an airlock inside [user], causing [user.p_them()] to explode!</span>")
+		user.gib()
+		return OBLITERATION
+
+	else														// Very shameful to end up here... Not enough ammo to use on self.
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:182]: I think you can just remove this else?

Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm:182]: Yep.

File: code/game/objects/items/weapons/RCD.dm (hunk 16 above)
Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm:145]: Moved.
Why 6,1 seconds (this just yeets after 1 second)?

Comment (DGamerL) [reply code/game/objects/items/weapons/RCD.dm]: So the animation has enough time to play out

EDIT: the RCD deleted it aswell it seems, up to you how you want to do this

Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm]: > So the animation has enough time to play out
> 
> EDIT: the RCD deleted it aswell it seems, up to you how you want to do this

The visible portion of the animation is only 5 seconds long. The rest of it is invisible. I have visually compared the suicide with 5 and 6 seconds and 6 seconds creates a noticeable delay between the animation appearing to end and the dusting happening.

Comment (DGamerL) [reply code/game/objects/items/weapons/RCD.dm]: As discussed in discord, fine with me

File: code/game/objects/items/weapons/RCD.dm [hunk 18]
```diff
@@ -131,6 +131,61 @@
...
+		playsound(loc, usesound, 50, TRUE)
+		for(var/obj/item/W in user)	// Do not delete all their stuff.
+			user.unEquip(W)			// Dump everything on the floor instead.
+		flags &= ~NODROP			// NODROP must be removed so the RCD doesn't get dusted along with them. Having this come after the unequipping puts the RCD on top of the pile of stuff (held items fall to the floor when dusting).
+		user.dust()
+		return OBLITERATION
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	if(mode == MODE_TURF && checkResource(3, user))		// Check you can afford to build this.
+		mode_turf(suicide_tile, user)					// Sart building the structure.
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:159]:
```suggestion
		mode_turf(suicide_tile, user)					// Start building the structure.
```

File: code/game/objects/items/weapons/RCD.dm [hunk 19]
```diff
@@ -131,6 +131,61 @@
...
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='notice'>Deconstructing user...</span>")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))					// Deconstruction of most structures takes 5 seconds, wait for the animation to finish.
+			QDEL_NULL(suicide_A)
+			return
+		if(!(user.l_hand == src) && !(user.r_hand == src))	// Do not commit die if the RCD isn't in your hands.
+			return SHAME								// If this triggers, someone probably chopped off your hand mid-suicide for some reason.
+		useResource(5, user)	// Consume ammo.
+		user.visible_message("<span class='suicide'>[user] deconstructs [user.p_they()]self with [src]!</span>")
```
Comment (Henri215) [inline code/game/objects/items/weapons/RCD.dm:149]: p_they() is the one that should be used here? Tested the PR today and i think i saw something odd with this message

Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm]: changed to p_themselves()

**Review by DGamerL — CHANGES_REQUESTED** [review]
I'm gonna have to ask to revert the RCD refactor for this PR, funny has a separate PR for this open already, and that way we can take a better look at the code and review it

**Comment by FunnyMan3595** [comment]
It's really a question of which PR we want to merge first. If we want #25062 first, than this one including it is fine. If we want the suicide act first, then I should add that to the refactor.

**Comment by Fordoxia** [comment]
> This pull request seems to be stale as there have been no changes in 14 days, please make changes within 7 days or the PR will be closed. If you believe this is a mistake, please inform a development team member on Discord.

Go away bot! Waiting for Funnyman PR!

**Comment by Henri215** [comment]
RCD refactor PR has been merged, have fun

File: code/game/objects/items/weapons/RCD.dm [hunk 20]
```diff
@@ -204,6 +204,20 @@
...
+	delay = 5 SECONDS
+	start_effect_type = /obj/effect/temp_visual/rcd_effect/reverse
+
+/datum/rcd_act/remove_user/can_act(atom/A, obj/item/rcd/rcd, mob/user)
+	if(!..())
+		return FALSE
+	return user.suiciding
+
+/datum/rcd_act/remove_user/act(atom/A, obj/item/rcd/rcd, mob/user)
+
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:220]: Bad merge master?

File: code/game/objects/items/weapons/RCD.dm [hunk 21]
```diff
@@ -50,12 +50,12 @@
 	return TRUE
 
 /// Test to see if the act is possible. You should usually override this.
-/datum/rcd_act/proc/can_act(atom/A, obj/item/rcd/rcd)
+/datum/rcd_act/proc/can_act(atom/A, obj/item/rcd/rcd, mob/user)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:53]:
```suggestion
/datum/rcd_act/proc/can_act(atom/A, obj/item/rcd/rcd)
```

File: code/game/objects/items/weapons/RCD.dm [hunk 22]
```diff
@@ -50,12 +50,12 @@
...
 	SHOULD_CALL_PARENT(TRUE)
 	return rcd.mode == mode
 
 /// Perform the act. You should usually override this.
-/datum/rcd_act/proc/act(atom/A, obj/item/rcd/rcd, mob/user)
+/datum/rcd_act/proc/act(atom/A, obj/item/rcd/rcd)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:58]: Bad merge master?

**Review by DGamerL — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/RCD.dm (hunk 22 above)
Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm:58]: *deathgasp

**Review by FunnyMan3595 — APPROVED** [review]

**Review by DGamerL — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/RCD.dm [hunk 23]
```diff
@@ -204,6 +204,18 @@
...
+	cost = 5
+	start_message = "Deconstructing user..."
+	delay = 5 SECONDS
+	start_effect_type = /obj/effect/temp_visual/rcd_effect/reverse
+
+/datum/rcd_act/remove_user/can_act(atom/A, obj/item/rcd/rcd)
+	if(!..())
+		return FALSE
+	return TRUE
+
```
[the remark this answers is not in the document]
Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm]: Doesn't seem to be a diff, removed.

**Review by Burzah — APPROVED** [review]

File: code/game/objects/items/weapons/RCD.dm [hunk 24]
```diff
@@ -331,6 +339,33 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON)
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		var/datum/rcd_act/remove_user/act = new
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:348]:
```suggestion
		var/datum/rcd_act/remove_user/act = new()
```

File: code/game/objects/items/weapons/RCD.dm [hunk 25]
```diff
@@ -331,6 +339,33 @@
...
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	if(afterattack(suicide_tile, user, TRUE))
+		user.visible_message("<span class='suicide'>[user] explodes as [src] builds a structure inside [user.p_them()]!</span>")
+		user.gib()
+		flags &= ~NODROP  // Ensure the RCD doesn't stick to the next person's hand.
+		return OBLITERATION
+
+	flags &= ~NODROP  // Ensure the RCD doesn't stick to the next person's hand.
+	return SHAME
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:367]:
```suggestion
	flags &= ~NODROP  // Ensure the RCD doesn't stick to the next person's hand.
	if(afterattack(suicide_tile, user, TRUE))
		user.visible_message("<span class='suicide'>[user] explodes as [src] builds a structure inside [user.p_them()]!</span>")
		user.gib()
		return OBLITERATION	
	return SHAME
```

File: code/game/objects/items/weapons/RCD.dm [hunk 26]
```diff
@@ -331,6 +339,31 @@
...
+		flags &= ~NODROP // If we got past the above check without dying, remove the superglue from the RCD.
+		return SHAME // Ensure that we do not accidentally perform the other flavour of suicide act.
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	flags &= ~NODROP  // Ensure the RCD doesn't stick to the next person's hand.
+	if(afterattack(suicide_tile, user, TRUE))
+		user.visible_message("<span class='suicide'>[user] explodes as [src] builds a structure inside [user.p_them()]!</span>")
+		user.gib()
+		return OBLITERATION	
+	return SHAME
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:365]:
```suggestion
	if(!afterattack(suicide_tile, user, TRUE))
		return SHAME
	user.visible_message("<span class='suicide'>[user] explodes as [src] builds a structure inside [user.p_them()]!</span>")
	user.gib()
	return OBLITERATION	
	
```

File: code/game/objects/items/weapons/RCD.dm [hunk 27]
```diff
@@ -331,6 +339,32 @@
...
+		var/datum/rcd_act/remove_user/act = new()
+		if(act.try_act(suicide_tile, src, user))
+			user.visible_message("<span class='suicide'>[user] deconstructs [user.p_themselves()] with [src]!</span>")
+			for(var/obj/item/W in user)	// Do not delete all their stuff.
+				user.unEquip(W)			// Dump everything on the floor instead.
+			flags &= ~NODROP			// NODROP must be removed so the RCD doesn't get dusted along with them. Having this come after the unequipping puts the RCD on top of the pile of stuff (held items fall to the floor when dusting).
+			user.dust()
+			return OBLITERATION
+		flags &= ~NODROP // If we got past the above check without dying, remove the superglue from the RCD.
+		return SHAME // Ensure that we do not accidentally perform the other flavour of suicide act.
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:357]: Could you also make this into an early return?

Comment (Fordoxia) [reply code/game/objects/items/weapons/RCD.dm]: It is done.

File: code/game/objects/items/weapons/RCD.dm [hunk 28]
```diff
@@ -331,6 +339,33 @@
...
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	if(!afterattack(suicide_tile, user, TRUE))
+		flags &= ~NODROP  
+		return SHAME
+	user.visible_message("<span class='suicide'>[user] explodes as [src] builds a structure inside [user.p_them()]!</span>")
+	flags &= ~NODROP  
+	user.gib()
+	return OBLITERATION	
+	
+
```
Comment (DGamerL) [inline code/game/objects/items/weapons/RCD.dm:368]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]

**Pull request merged**